### PR TITLE
Add support for ReSignIn

### PR DESCRIPTION
**Pull request by HaoK**
- Add ResignIn api to regenerate the application cookie preserving rememberMe and AuthenticationMethod
- Update manage controller in samples to use new API

Fixes: https://github.com/aspnet/Identity/issues/185

@divega @Tratcher @suhasj @rustd @blowdart

### CODE REVIEW

File: src/Microsoft.AspNet.Identity/SignInManager.cs [hunk 1]
```diff
@@ -75,6 +75,21 @@ public class SignInManager<TUser> where TUser : class
...
+        /// Called to regenerate the ApplicationCookie for the user, preserving the existing
+        /// AuthenticationProperties like rememberMe
+        /// </summary>
+        /// <param name="user"></param>
+        /// <returns></returns>
+        public virtual async Task ResignInAsync(TUser user)
+        {
+            var authResult = await Context.AuthenticateAsync(IdentityOptions.ApplicationCookieAuthenticationScheme);
+            var properties = authResult?.Properties;
+            var isPersistent = properties != null && properties.IsPersistent;
```
Comment (suhasj) [inline src/Microsoft.AspNet.Identity/SignInManager.cs]: So we are preserving only the `IsPersistent` property ? The other property I feel might be slightly important is the Dictionary. Do you think we need that too ?

Comment (HaoK) [inline src/Microsoft.AspNet.Identity/SignInManager.cs]: @Tratcher what do you think we should preserve when it comes to the AuthenticationProperties, we definitely want to regenerate the claims, but we want to keep as many of the old settings as possible that make sense...

Comment (Tratcher) [inline src/Microsoft.AspNet.Identity/SignInManager.cs]: It's a generic property bag, you'd want to preserve anything you can.

Comment (HaoK) [inline src/Microsoft.AspNet.Identity/SignInManager.cs]: Is it safe to copy everything over then?  Except for maybe nulling out ExpiresUtc and IssuedUtc unless SignIn will take care of restamping those for us...

Comment (HaoK) [inline src/Microsoft.AspNet.Identity/SignInManager.cs]: @divega We can consider changing SignIn to mimic the underlying SignIn API more closely, remove IsPersistent and take in a raw AuthenticationProperties which would expose everything

**Comment by HaoK** [comment]
Updated, kept old SignIn overload which just takes IsPersistence, but added an overload which takes AuthenticationProperties, and switched ReSignInAsync to use that.

**Comment by divega** [comment]
I am ok with this, but didn't you say that we were the main users of this API and it didn't make sense to keep the overload that takes a bool? Anything changed your point of view?

**Comment by HaoK** [comment]
Well, I tried it without the overload, and it was annoying and i immediately wrote a helper overload, so might as well keep it and leave it public (no breaking change at all this way too).  So that is what convinced me to keep it :)

**Comment by divega** [comment]
Fair enough :smile: :shipit:

**Comment by rustd** [comment]
The name of the method is not great. How about RefreshIdentity? RegenerateIdentity?

**Comment by divega** [comment]
Fair point too. Seems to me that the word "renew" may be closer in meaning to what the API will do.

**Comment by divega** [comment]
E.g. 'RenewSignIn' or something like that.

**Comment by HaoK** [comment]
RenewSignIn or RefreshSignIn works for me

**Comment by rustd** [comment]
Either one is good. Let's pick RefreshSignIn?

**Comment by HaoK** [comment]
Ok updated to RefreshSignIn

**Comment by divega** [comment]
:ship it:

**Comment by HaoK** [comment]
c251bf0162091730b6cc74d97731db74d98e499d

**Pull request closed without merging**